src/quo2/components/bottom_sheet/view.cljs

```diff
@@ -0,0 +1,196 @@
+(ns quo2.components.bottom-sheet.view
+  (:require [oops.core :refer [oget]]
+            [quo.react :as react]
+            [quo2.components.bottom-sheet.styles :as styles]
+            [re-frame.core :as re-frame]
+            [react-native.background-timer :as timer]
+            [react-native.core :as rn]
+            [react-native.gesture :as gesture]
+            [react-native.hooks :as hooks]
+            [react-native.platform :as platform]
+            [react-native.reanimated :as reanimated]
+            [react-native.safe-area :as safe-area]
+            [reagent.core :as reagent]))
+
+(def bottom-sheet-js (js/require "../src/js/bottom_sheet.js"))
+
+(def animation-delay 450)
+
+(defn with-animation [value & [options callback]]
+  (reanimated/with-spring
+    value
+    (clj->js (merge {:mass 2
+                     :stiffness 500
+                     :damping 200})
+             options)
+    callback))
+
+(def content-height (reagent/atom nil))
+(def show-bottom-sheet? (reagent/atom nil))
+(def keyboard-was-shown? (reagent/atom false))
+(def expanded? (reagent/atom false))
+(def gesture-running? (reagent/atom false))
```

Comment:
I am not sure if they all need to be reagent atoms. If we handle animation at the source (without effects), then we only care about value, not react to them. (We probably can do it later)